yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,129 @@
+            type: string
+            description: >
+                Inventory/hardware Path of the FRU whose keyword needs to be
+                read.
+          - name: data
+            type: variant[struct[string,string],struct[string]]
+            description: >
+                Contains data required to read. currently supports IPZ and
+                Keyword type VPD and more can be added.
+          - name: target
+            type: boolean
+            description: >
+                Default will be false and will read from hardware in that case,
+                if true will read from cache.
+      returns:
+          - name: value
+            type: array[byte]
+            description: >
+                Value of the given keyword read from the given target.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: GetLocationCode
+      description: >
+          Method to get the expanded location code for the given inventory path.
+          If expanded location code is not available then unexpanded location
+          code will be returned.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                If expanded location code is available it returns expanded
+                location code else returns unexpanded location code.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - com.ibm.VPD.Error.PathNotFound
+
+    - name: deleteFRUVPD
+      description: >
+          Method to delete given FRU VPD and set its present property as false.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: CollectFRUVPD
+      description: >
+          Method to perform VPD collection for the given FRU. This method
+          requires to have JSON defined. The FRU should have concurrently
+          Maintainable flag set to  true in VPD JSON to execute this action.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU whose VPD needs to be collected.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: PerformVPDRecollection
+      description: >
+          Method to check for FRU's replaced and if required, perform VPD
+          recollection for them.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+properties:
+    - name: VPDCollectionStatus
+      type: enum[self.VPDStatus]
+      default: NotStarted
+      description: >
+          Property which provides the status of FRU's VPD collection.
```

Comment:
should we mention- system's all FRU's VPD collection. ?

Reply:
I think it is not required

Reply:
The description is not correct. This does not denotes FRUs VPD collection status.

Reply:
updated